articles/javascript-dates-manipulation-with-date-fns/index.md

```diff
@@ -0,0 +1,475 @@
+
+```js
+const {format} = require('date-fns');
+//parse a default date value
+const date = new Date('2020/09/19');
+console.log(date);
+console.log(`${format(date, 'dd.MM.yyyy')}`);
+console.log(`${format(date, 'yyyy-MM-dd').toString()}`);
+console.log(`${format(date, 'EEEE, MMMM yyyy')}`);
+console.log(`${format(date, 'EEEE,MMMM do, yyyy hh:mm a')}`);
+console.log(`${format(date, 'MMMM, yyyy')}`);
+console.log(`${format(date, 'MMMM.do.')}`);
+console.log(`${format(date, 'EEEE do HH:mm ')}`);
+console.log(`${format(date, 'EEEE,MMMM do, yyyy ppppp')}`);
+console.log(`${format(date, 'do  MMMM yyyy OOOO')}`);
+```
+The above examples will display the date parsed with several date formats. [Check out more date formats](https://date-fns.org/v2.16.1/docs/format) you can play with and get more ideas of how date-fns formats works.
+
+**Note:** we have only imported the `format` function as it is what we need for the moment, that's one of the dynamic features of date-fns. When formatting date-fns token values try to avoid some [common mistakes](https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md) such as:
+
+```js
+format(date, 'DD.MM.YYYY'); ❌
+format(date, 'dd.MM.yyyy'); ✔
+```
+
+### Date Arithmetic (additions, subtractions)
+It is hard to do arithmetic calculations for dates. Date-fns simplifies addition and subtraction of dates between years, months, days, weeks and seconds using simple functions such as `addDays`, `addWeeks`, `addMonths`, `subDays`, `subWesks`, `subMonths` etc.
+
+#### Additions
+Syntax
+`addDays(date, amount)`
+- date - The date to be changed
+- amount - Amount of days to be added
+
+Let's perform simple date [addition](https://date-fns.org/v2.16.1/docs/add). To get started, import the `add` functions, then add the unit of time to the base date. Specify the operation you want to perform as the first argument followed by the number of units to add. Include the format function to format the date returned.
+
+Example:
+
+<iframe height="400px" width="100%" src="https://repl.it/@kimkimani/date-fns-add?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+**Note:** date units added/subtracted with positive decimals will be rounded off with `math.floor` and decimals less than zero will be rounded using `math.cell`.
+
+#### Subtractions
+Works exactly like addition, only that the `add` prefix function is replaced with a `sub`. Then [subtract](https://date-fns.org/v2.16.1/docs/sub) your specified units of time.
+
+Syntax example:
+`subDays(date, amount)`
+Import sub function as shown in the example below. Similarly, you can choose format function to manipulate your display options.
+```js
+const date = new Date('2020.09.29 10:12:00');
+//sub days
+const a = subDays(date, 4);
+//sub months
+const b = subMonths(date, 3);
+//sub years
```

Comment:
Maybe this is more of a technical issue but I think there is a missing line of code. The `sub years` part.